src/metabase/driver/postgres.clj

```diff
@@ -362,48 +371,69 @@
 
 (defmethod sql.qp/datetime-diff [:postgres :second]
   [_driver _unit x y]
-  (let [seconds (hx/- (extract :epoch y) (extract :epoch x))]
+  (let [seconds (hx/- (extract-from-timestamp :epoch y) (extract-from-timestamp :epoch x))]
     (hx/->integer (hx/call :trunc seconds))))
 
-(p/defrecord+ RegexMatchFirst [identifier pattern]
-  hformat/ToSql
-  (to-sql [_]
-    (str "substring(" (hformat/to-sql identifier) " FROM " (hformat/to-sql pattern) ")")))
+(defn- format-regex-match-first [_fn [identifier pattern]]
+  (let [[identifier-sql & identifier-args] (sql/format-expr identifier {:nested true})
+        [pattern-sql & pattern-args]       (sql/format-expr pattern {:nested true})]
+    (into [(format "substring(%s FROM %s)" identifier-sql pattern-sql)]
+          cat
+          [identifier-args
+           pattern-args])))
+
+(sql/register-fn! ::regex-match-first #'format-regex-match-first)
```

Comment:
Here's an example of converting a custom clause from Honey SQL 1 to Honey SQL 2.

This stuff is not used by application database code so it's ok to remove the Honey SQL 1 version entirely.